api/src/resources/subscription/subscription.service.ts

```diff
@@ -25,6 +26,9 @@ const updateSubscription = async (data: any) => {
       upsert: true,
     },
   );
+  if (data.cancel_at_period_end) {
```

Comment:
Every single subscription update event will trigger this if subscription has `cancel_at_period_end=true`